server.js

```diff
@@ -104,7 +104,26 @@ async function getComponentMdxSource(params, swingsetOptions) {
     fs.readFileSync(currentComponentData.propsPath, 'utf8')
 
   // Inject a primary headline with the component's name above the content
-  const contentWithHeadline = `# \`<${data.componentName}>\` Component\n${content}`
+  let contentWithHeadline = `<div className='__swingset-headline'>
+  <h1><code>&lt;${data.componentName}&gt;</code> Component</h1>`
+
+  // If custom metadata is called for, add it below the headline
+  if (swingsetOptions.customMeta) {
+    const customMeta = swingsetOptions.customMeta(currentComponentData)
+    contentWithHeadline += '\n<div className="__swingset-meta">'
+
+    if (customMeta.github) {
+      contentWithHeadline += `<a className='__swingset-meta-github' target='_blank' rel='noopener' href='${customMeta.github}' title='View Source on GitHub'></a>`
+    }
+
+    if (customMeta.npm) {
+      contentWithHeadline += `<a className='__swingset-meta-npm' target='_blank' rel='noopener' href='${customMeta.npm}' title='View NPM Package'></a>`
+    }
+
+    contentWithHeadline += '\n</div>'
+  }
```

Comment:
I'm not sure if this would be better, but did you think about using a custom component here and injecting that and the data?

I _think_ it might look like this:
```jsx
const metaObject = swingsetOptions.customMeta ? swingsetOptions.customMeta(currentComponentData) : {}

let contentWithHeadline = `<div className='__swingset-headline'>
  <h1><code>&lt;${data.componentName}&gt;</code> Component</h1>
<CustomMeta meta={${JSON.stringify(metaObject, null, 2)} />`
```

the `CustomMeta` component would then contain the markup and we'd have to wire it through in `page.jsx`.


Food for 💭

Reply:
yeah i did actually consider this but then just didn't do it haha. i will give that a shot though and see if it works and can clean things up

Reply:
Possibly related: I'm [down a bit of a "default template" rabbit hole](https://app.asana.com/0/1100423001970639/1200426948419712/f). If `CustomMeta` was provided in a similar way to `LiveComponent`, it feels like a short (admittedly not-in-the-scope-of-this-PR) hop to allow custom templates, which might remove the need for `swingsetOptions.customMeta` altogether (as the consumer could construct it from the `packageJson` data they already have access to). Eg something like:

```js
const mdxPrefix = swingsetOptions.componentMdxTemplate ?? `<div className='__swingset-headline'>
  <h1><code>&lt;${data.componentName}&gt;</code> Component</h1>`

let contentWithHeadline = `${templateMdx}${content}`
```

And then in a consumer, eg `react-components`:

```js
const swingsetConfig = {
  componentMdxTemplate: "<div className='__swingset-headline'>\n<h1><code>&lt;${data.componentName}&gt;</code> Component</h1>\n<CustomMeta meta={{ github: `https://github.com/hashicorp/react-components/tree/main/packages/${data.slug}`, npm: `https://npmjs.com/package/${packageJson.name}` }} />"
}
```

(Would be ideal if the above was like a file path to `template.mdx` instead of string literal, but hopefully the idea comes through!)

Reply:
This is a really great point, and I love the line of thought around templates, that would be an amazing feature!